brigade/plugins/inventory/ansible.py

```diff
@@ -0,0 +1,193 @@
+            self.hosts[host].update(self.read_vars_file(host, self.path, True))
+            self.hosts[host] = self.map_brigade_vars(self.hosts[host])
+
+    def sort_groups(self):
+        for host in self.hosts.values():
+            host["groups"].sort()
+
+        for name, group in self.groups.items():
+            if name == "defaults":
+                continue
+
+            group["groups"].sort()
+
+    def read_vars_file(self, element, path, is_host=True):
+        subdir = "host_vars" if is_host else "group_vars"
+        filepath = os.path.join(path, subdir, element)
+
+        if not os.path.exists(filepath):
+            logger.debug(
+                "AnsibleInventory: var file doesn't exist: {}".format(filepath)
+            )
+            return {}
+
+        with open(filepath, "r") as f:
+            logger.debug("AnsibleInventory: reading var file: {}".format(filepath))
+            yml = ruamel.yaml.YAML(typ="rt", pure=True)
+            return yml.load(f)
+
+    def map_brigade_vars(self, obj):
+        mappings = {
+            "ansible_host": "brigade_host",
+            "ansible_port": "brigade_ssh_port",
+            "ansible_user": "brigade_username",
+            "ansible_password": "brigade_password",
+        }
+        result = {}
+        for k, v in obj.items():
+            if k in mappings:
+                result[mappings[k]] = v
+            else:
+                result[k] = v
+        return result
+
+    def add(self, element, element_dict):
+        if element not in element_dict:
+            element_dict[element] = {"groups": []}
+
+
+class INIParser(AnsibleParser):
+
+    def normalize_content(self, content):
+        result = {}
+
+        if not content:
+            return result
+
+        for option in content.split():
+            k, v = option.split("=")
+            try:
+                v = int(v)
+            except Exception:
+                pass
+            result[k] = v
+
+        return result
+
+    def normalize(self, data):
+        result = {}
+        for k, v in data.items():
+            meta = None
+            if ":" in k:
+                k, meta = k.split(":")
+            if k not in result:
+                result[k] = {}
+
+            if meta not in result[k]:
+                result[k][meta] = {}
+
+            if meta == "vars":
+                for data, _ in v.items():
+                    result[k][meta].update(self.normalize_content(data))
+            elif meta == "children":
+                result[k][meta] = {k: {} for k, _ in v.items()}
+            else:
+                result[k]["hosts"] = {
+                    host: self.normalize_content(data) for host, data in v.items()
+                }
+        return result
+
+    def load_hosts_file(self):
+        hostsfile = cp.ConfigParser(
+            interpolation=None, allow_no_value=True, delimiters=" "
+        )
+        hostsfile.read(os.path.join(self.path, "hosts"))
```

Comment:
We shouldn't assume that the file is named `hosts`, it should be allowed to have any name.